### PR TITLE
fix(oans): use IRS position & display correct flags when it's not available ♻️

### LINKED ISSUE #9317: A380X/ND: OANS available without IRS position
### Aircraft Version

Development

### Build info

```json
master
```

### Describe the bug

OANS is available without IRS position:
image

### Expected behavior

OANS position should not be available if the on side ADIRS does not have a valid position. In this situation, the "ARPT NAV POS LOST" amber message should appear.
Captain side uses ADIRS1 while FO ADIRS2, however, it is possible to change the position data source to ADIRS3 to either side with the ATT HDG selector.

### Steps to reproduce

1) Spawn in with the IRS off and set the ND range to "ZOOM"

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

bruno_pt99

### PR DESCRIPTION
**Pull request by Nufflee**
Fixes #9317

## Summary of Changes
Previously, OANS was always shown regardless of IRS position availability (actually, it should be using GPS + IRS velocity data, but we don't simulate GPS yet). Now, when IRS position is not available, the "ARPT NAV POS LOST" flag is displayed in ROSE-NAV and ARC modes, while in PLAN mode OANS is available with the same flag at the bottom. Airport auto selection is also inhibited and the OANS is centered at the ARP by default when IRS is unavailable. The OANS flag display system was also refactored.

Alongside that, I fixed a bug where the OANS AMDB validity dates showed the day of the week instead of the day of the month and tweaked flag font size to better match real life. 

### ROSE-NAV without IRS
image

### ARC without IRS
image

### PLAN without IRS
image

## References

https://www.youtube.com/watch?v=OaqDFPNUZ04

image

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. In order to make Navigraph integration work, copy the `flybywire-aircraft-a380-842\html_ui\Pages\VCockpit\Instruments\A380X\EFB\efb.js` file from the A380X package installed _via the installer_ to the same place in the QA package from this PR
1. Spawn in cold and dark and energize the aircraft, but don't align IRS
1. Ensure that OANS is not available in ROSE-NAV and ARC modes but displays an empty screen in PLAN mode. The "ARPT NAV POS LOST" flag should be visible in all modes
1. In PLAN mode, select an airport and ensure it's displayed
1. Align IRS and ensure that all modes work as before and that the aircraft icon is displayed

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

### CODE REVIEW

File: fbw-common/src/systems/instruments/src/OANC/Oanc.tsx [hunk 1]
```diff
@@ -391,19 +438,20 @@ export class Oanc<T extends number> extends DisplayComponent<OancProps<T>> {
...
-          this.props.messageScreenRef.instance.style.visibility = 'hidden';
-          this.props.messageScreenRef.instance.innerText = '';
-          this.props.messageScreenRef.instance.classList.remove('amber');
-        }
-      }
-    }, true);
+    this.sub
+      .on('latitude')
+      .whenChanged()
+      .handle((v) => this.pposLatWord.setWord(v));
```
Comment (flogross89) [inline fbw-common/src/systems/instruments/src/OANC/Oanc.tsx:444]: You could also replace these these subscriptions by using `Arinc429LocalVarConsumerSubject` , e.g. something like `private readonly pposLatWord = Arinc429LocalVarConsumerSubject.create(this.sub.on('latitude'));`

**Review by flogross89 — CHANGES_REQUESTED** [review]
Thanks a lot! Looking good, just one minor remark.

File: fbw-common/src/systems/instruments/src/OANC/Oanc.tsx (hunk 1 above)
Comment (Saschl) [reply fbw-common/src/systems/instruments/src/OANC/Oanc.tsx:444]: Also, .whenChanged should be obsolete ;)

File: fbw-common/src/systems/instruments/src/OANC/Oanc.tsx [hunk 2]
```diff
@@ -61,6 +61,7 @@ import { OancLabelManager } from './OancLabelManager';
 import { OancPositionComputer } from './OancPositionComputer';
 import { NavigraphAmdbClient } from './api/NavigraphAmdbClient';
 import { globalToAirportCoordinates, pointAngle, pointDistance } from './OancMapUtils';
+import { GenericAdirsEvents } from 'instruments/src/ND/types/GenericAdirsEvents';
```
Comment (Saschl) [inline fbw-common/src/systems/instruments/src/OANC/Oanc.tsx:64]: I believe this import path does not work

**Review by flogross89 — APPROVED** [review]
Very cool changes, thanks! :)

**Comment by alepouna** [comment]
Quality Assurance Tester/Trainee Report

Discord Username  : alepouna 
Object of testing     : #9487 
Aircraft                    : A380X 
Tier of Testing          : 1
Date                        : 22/11/2024 

Testing Process: 
 
- Spawn in cold and dark and energize the aircraft, but don't align IRS
- Check OANS is not available in ROSE-NAV and ARC modes but displays an empty screen in PLAN mode. "ARPT NAV POS LOST" is visible in all modes [PASS]
- In PLAN mode, select an airport
- Check it's displayed [PASS]
- Align IRS and confirm that all modes work as before and that the aircraft icon is displayed [PASS]

Testing Results: 
Passed

**Pull request merged**